makes changes in Dismissible Widget

theankurkedia opened the pull request:
adds dismissDuration and diagonalDisplacement properties.
`dismissDuration` is used to define the dismiss time of card to springs back to original position
`diagonalDisplacement` is used to define the displacement of card from movement axis after it is swiped, works for both vertical and horizontal directions

Commit:
adds dismissDuration and diagonalDisplacement properties in Dismissible widget

Commit:
adds dismissDuration and diagonalDisplacement properties in Dismissible widget

theankurkedia commented:
I signed it!

xster commented on packages/flutter/lib/src/widgets/dismissible.dart:
```diff
@@ -134,6 +135,12 @@ class Dismissible extends StatefulWidget {
   final Map<DismissDirection, double> dismissThresholds;
 
+  final Duration dismissDuration;
+  /// defines the duration for card to dismiss and come to original position
```
Dart docs above the member being documented and in complete sentences (capitalize and period).

xster commented on packages/flutter/lib/src/widgets/dismissible.dart:
```diff
@@ -134,6 +135,12 @@ class Dismissible extends StatefulWidget {
+  final Duration dismissDuration;
+  /// defines the duration for card to dismiss and come to original position
+
+  final double diagonalDisplacement;
```
Seems specific. Is the desired feature just to be able to specify dismiss in an arbitrary vector?

theankurkedia replied:
The movement is along the across the direction. Otherwise, the movement is just along the direction. It makes the widget more customisable.

yjbanov committed:
mark complex_layout_scroll_perf_ios__timeline_summary as flaky\ (#14080)

srawlins committed:
Bump collection dependency to 1.14.5 (#14048)

Hixie committed:
Fix ShapeDecoration (#14009)

And add a test that catches these problems.

goderbauer committed:
Add Flutter Console for Windows (#14030)

Just download a Flutter ZIP bundle, unzip, and double-click on `flutter_console.bat` to get started. No PATH modifications required.

cbracken committed:
Decode syslog-encoded iOS logs (#14075)

Apple encodes syslog entries using a 7-bit encoding where input UTF-8 bytes
are encoded as follows:
1. 0x00 to 0x19: non-printing range. Some ignored, some encoded as <...>.
2. 0x20 to 0x7f: as-is, with the exception of 0x5c (backslash).
3. 0x5c (backslash): octal representation \134.
4. 0x80 to 0x9f: \M^x (using control-character notation for range 0x00 to 0x40).
5. 0xa0: octal representation \240.
6. 0xa1 to 0xf7: \M-x (where x is the input byte stripped of its high-order bit).
7. 0xf8 to 0xff: unused in 4-byte UTF-8.

As there doesn't appear to be a system tool to decode these strings, we
implement here in Dart. If we're unable to decode a string (e.g.
decoding results in an invalid UTF-8 string), we fall back to emitting
the log line as-is.

Hixie commented:
This needs a test.

Can you talk more about diagonalDisplacement? (Maybe add some more details to the documentation?) The feature seems fine but maybe the name should be different. I'm not sure the name really conveys what it does clearly.

Hixie commented:
(The test should verify what it is that you want to make sure we don't regress, because without tests, it's quite possible we'd just remove the feature thinking it was unused!)

amirh committed:
Add a PhysicalShape widget and a render object for it. (#13682)

This CL also refactors common logic for RenderPhysicalModel and
RenderPhysicalShape into a base class _RenderPhysicalModelBase.

srawlins committed:
Bump json_rpc_2 dependency (#14051)

yjbanov committed:
flutter_view_ios__start_up: move all test logic into TaskFunction (#14088)

Commit:
TextField splash integration (#14055)

jason-simmons committed:
Use a safe codec to read Skia SKP screenshot files when checking for errors (#14086)

If the downloaded file is an actual SKP and not an error report, then the
default UTF-8 codec will fail to decode the SKP content

mraleph committed:
Fix a couple of strong mode issues. (#14070)

* JSON.decode produces Map<String, dynamic> and List<dynamic>
objects. If a more tight type is required then object needs to
be converted explicitly (see dart-lang/sdk#31876);
* Completer<dynamic> produces Future<dynamic>. In Dart 2 it is
runtime error to assign Future<dynamic> to variable of type Future<T>;

Commit:
Changed 'dismissDuration'to 'rollbackDuration' and 'diagonalDisplacement' to 'crossAxisEndOffset' and added details

Commit:
renamed 'dismissDuration' with 'rollbackDuration' and 'diagonalDisplacement' with 'crossAxisEndOffset' and added details

Commit:
renamed 'dismissDuration' with 'rollbackDuration' and 'diagonalDisplacement' with 'crossAxisEndOffset' and added details

theankurkedia commented:
@Hixie I have renamed `diagonalDisplacement` with `crossAxisEndOffset` and `dismissDuration` with `rollbackDuration` and added documentation also.

Hixie commented:
Nevermind, I see https://github.com/flutter/flutter/pull/14162 now.

theankurkedia commented:
I decided to close this because I accidentally through fork, so I could not get the CLA. As you have already noticed, I have created a new pull request.
